core/models.py

```diff
@@ -268,6 +279,57 @@ def save(self: "PostVote", *args: int, **kwargs: int) -> None:
         Post.objects.filter(pk=self.post.pk).update(up_votes=up_votes, down_votes=down_votes)
 
 
+class PostAward(models.Model):
+    REWARD_POINTS: ClassVar[dict[int, int]] = {
+        1: 15,  # Level 1 gives 15 points
+        2: 25,  # Level 2 gives 25 points
+        3: 50,  # Level 3 gives 50 points
+    }
+
+    REWARD_CHOICES: ClassVar[list[tuple[str, str]]] = []
+
+    for level, points in REWARD_POINTS.items():
+        for i in range(1, 6):  # 5 icons for each level
+            reward_code = f"{level}{i}_REWARD"
+            reward_points = f"{points} points"
+            REWARD_CHOICES.append((reward_code, reward_points))
+
+    user = models.ForeignKey(User, on_delete=models.CASCADE, related_name="awards")
+    post = models.ForeignKey(Post, on_delete=models.CASCADE, related_name="post_awards")
+    choice = models.CharField(max_length=20, choices=REWARD_CHOICES)
+    gold = models.IntegerField(default=0)
+    created_at = models.DateTimeField(auto_now_add=True)
+    anonymous = models.BooleanField(default=False)
+    comment = models.CharField(max_length=100, blank=True, default="")
+
+    class Meta:
+        unique_together: ClassVar[list[str]] = ["post", "user"]
+
+    def __str__(self: "PostAward") -> str:
+        return f"@{self.user}: {self.choice} for post: {self.post}"
+
+    def save(self: "PostAward", *args: int, **kwargs: int) -> None:
+        if self.check_duplicate_award():
+            raise ValueError("Already given award")

```

Comment:
Won't it already fail on unique together?
Also the check should probably be in the form as I am not sure it will not throw 500 instead of nice form error. Please double check.